src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/component/CommunityAPIClient.kt

```diff
@@ -18,9 +19,35 @@ class CommunityAPIClient(
       .retrieve()
       .bodyToMono(Unit::class.java)
       .onErrorResume { e ->
-        logger.error("Call to community api failed", e, StructuredArguments.kv("requestBody", requestBody))
+        handleResponse(e, requestBody)
         Mono.empty()
       }
       .subscribe()
   }
+
+  fun <T : Any> makeSyncPostRequest(uri: String, requestBody: Any, responseBodyClass: Class<T>): T? {
```

Comment:
nice - you can also use a reified type param to avoid having to pass the class in the method call

Reply:
e.g. ```

  inline fun <reified T> makeSyncPostRequest(uri: String): T? {
    ...bodyToMono(T)
  }
  ```

Reply:
Love it